e2e/tests/eth_integration.rs

```diff
@@ -0,0 +1,273 @@
+
+    // Approve GPv2 for trading
+    tx!(trader_buy_eth_a, token.approve(gpv2.allowance, to_wei(50)));
+    tx!(trader_buy_eth_b, token.approve(gpv2.allowance, to_wei(50)));
+    tx!(trader_sell_weth, weth.approve(gpv2.allowance, to_wei(100)));
+
+    let native_token = weth.address();
+    let OrderbookServices {
+        orderbook,
+        price_estimator,
+    } = OrderbookServices::new(&web3, &gpv2, &uniswap_factory, native_token).await;
+
+    let client = reqwest::Client::new();
+
+    // Test fee endpoint
+    let client_ref = &client;
+    let estimate_fee = |sell_token, buy_token| async move {
+        client_ref
+            .get(&format!(
+                "{}{}?sellToken={:?}&buyToken={:?}&amount={}&kind=sell",
+                API_HOST,
+                FEE_ENDPOINT,
+                sell_token,
+                buy_token,
+                to_wei(42)
+            ))
+            .send()
+            .await
+            .unwrap()
+    };
+    let fee_with_weth = estimate_fee(token.address(), BUY_ETH_ADDRESS).await;
+    assert_eq!(fee_with_weth.status(), 200);
+    // Eth is only supported as the buy token
+    let fee_with_eth = estimate_fee(BUY_ETH_ADDRESS, token.address()).await;
+    assert_eq!(fee_with_eth.status(), 404);
+
+    // Place Orders
+    assert_ne!(weth.address(), BUY_ETH_ADDRESS);
+    let order_buy_eth_a = OrderBuilder::default()
+        .with_kind(OrderKind::Buy)
+        .with_sell_token(token.address())
+        .with_sell_amount(to_wei(50))
+        .with_buy_token(BUY_ETH_ADDRESS)
+        .with_buy_amount(to_wei(49))
+        .with_valid_to(shared::time::now_in_epoch_seconds() + 300)
+        .with_signing_scheme(SigningScheme::Eip712)
+        .sign_with(
+            &gpv2.domain_separator,
+            SecretKeyRef::from(&SecretKey::from_slice(&TRADER_BUY_ETH_A_PK).unwrap()),
+        )
+        .build()
+        .order_creation;
+    let placement = client
+        .post(&format!("{}{}", API_HOST, ORDER_PLACEMENT_ENDPOINT))
+        .body(json!(order_buy_eth_a).to_string())
+        .send()
+        .await;
+    assert_eq!(placement.unwrap().status(), 201);
+    let order_buy_eth_b = OrderBuilder::default()
+        .with_kind(OrderKind::Sell)
+        .with_sell_token(token.address())
+        .with_sell_amount(to_wei(50))
+        .with_buy_token(BUY_ETH_ADDRESS)
+        .with_buy_amount(to_wei(49))
+        .with_valid_to(shared::time::now_in_epoch_seconds() + 300)
+        .with_signing_scheme(SigningScheme::Eip712)
+        .sign_with(
+            &gpv2.domain_separator,
+            SecretKeyRef::from(&SecretKey::from_slice(&TRADER_BUY_ETH_B_PK).unwrap()),
+        )
+        .build()
+        .order_creation;
+    let placement = client
+        .post(&format!("{}{}", API_HOST, ORDER_PLACEMENT_ENDPOINT))
+        .body(json!(order_buy_eth_b).to_string())
+        .send()
+        .await;
+    assert_eq!(placement.unwrap().status(), 201);
+
+    let order_sell_eth = OrderBuilder::default()
```

Comment:
nit: What extra bit is this testing? Would it suffice to only buy eth with a single order and rely on the AMM to provide the matching? This might make the test significantly shorter?